### DISCUSSION TITLE
PF4 : ICP3 !

### DISCUSSION THREAD

**Discussion by trimarco232** [dbuezas/lgt8fx#277] [Ideas]
there is a thing that is interesting for who needs a second capture input, that does not exist on atmega328p
typically if one want to decode manchester like signals , or uart , for witch it is probably better  using ICP3 , because of the digital filter
ICP3 is on pin PF4 ; because PF4 is pin 3 of the 32 pins chips , witch is connected to gnd on the mini or nano boards , only the 48 pins board has it available
really ? according to the datasheet, port PF4 is connected to pin 3 of the 20 pins chip , together wit PD3 and PD4
maybe it worth the effort to mention it (or complete the drawings)

**Comment by dbuezas** [comment]
Indeed! @wollewald made an update today, but I think this is not included.

It's not shown In the QFP32 chip pinout, but it is there in the table
<img width="741" alt="image" src="https://github.com/dbuezas/lgt8fx/assets/777196/5c845c53-f564-441d-a9b5-fd7adbb6bc09">
<img width="403" alt="image" src="https://github.com/dbuezas/lgt8fx/assets/777196/0534869b-c902-49be-911a-6fcfe19ee8d4">
<img width="365" alt="image" src="https://github.com/dbuezas/lgt8fx/assets/777196/0c544cb2-c85b-482a-8a55-02d06e799dd5">
<img width="339" alt="image" src="https://github.com/dbuezas/lgt8fx/assets/777196/66d98104-13b8-44ec-b459-5fe82a727ec2">

**Reply by wollewald** [reply]
@dbuezas, @trimarco232, yes, seems I forgot the ICP3 on the pinout of the Pro Mini style SSOP20. Will add it.

But I am not sure where to find ICP3 on the QFP32L. The screenshot of dbuezas suggests it is at physical Pin3 (= PE4) of the IC itself. But PE4 is not connected to any board pin of the Nano style and Pro Mini style QFP32L boards. And PF4 does not exist as it seems.

Does anyone know for sure where it is on the QFP32L?

**Reply by dbuezas** [reply]
It looks like it is not connected to any pin: 
![image](https://github.com/dbuezas/lgt8fx/assets/777196/0c1de072-5e1a-4df0-b890-d5e8813eab72)

**Reply by dbuezas** [reply]
Oh wrong direction. Anyway still not connected. I guess one can still solder a wire to the pad, but that's beyond your fantastic pinout diagrams

**Reply by trimarco232** [reply]
Hi,
PF4 is ICP3 and OC1B
unfortunately , none of the pin#3 (PE4+PF4) nor the pin#6 (PE5+PF5) of the 32pin chip is connected to any pin of the nano or mini boards
this is because pin#3 and pin#6 of the atmega328p are connected to gnd
(may I suggest our Chinese friends to remake these boards and connect these pins instead of the two rst pins , witch have no utility)

**Reply by dbuezas** [reply]
Which would be particularly useful b/c f4 and f5 can be high current outputs

**Reply by wollewald** [reply]
I added ICP3. I also "found" PF4 and PF7 and added them. A pull request is coming. Feels like an asymptotic approximation to the truth.

**Reply by dbuezas** [reply]
Asymptotic indeed 😂

**Comment by trimarco232** [comment]
Hi, PD2 is analog AC0O (comparator 0 output)
this allows the complete access to the comparator , in combinaison with the pins AC0P and ACXN
(still few space between the 2 curves 😂)

**Reply by wollewald** [reply]
Why am I so blind? It seems I have worked too long with these pinout diagrams! But thanks a lot. At least a bit of the space between the curves will be filled. I will send the next PR in a few minutes.
